### ISSUE TITLE
Evaluating discriminator accuracy

### ISSUE THREAD

**Issue by slala2121** [mit-han-lab/data-efficient-gans#54]
How did you measure the discriminator training/validation accuracy (as in Fig. 6 from the paper)? When I evaluated on the real images, I find that the discriminator generally classifies them as fake. On the synthesized images, the discriminator classifies them as fake as well.

Below is an example of how I'm computing the accuracy on the real data: 

`with tf.device('/gpu:0'):
            print('Constructing networks...')
            
            D_neg = tflib.Network('D', num_channels=num_channels, resolution=resolution, label_size=label_size, **D_args)
            resume_networks = misc.load_pkl(resume_pkl_neg)
            rG, rD, rGs = resume_networks
            D_neg.copy_vars_from(rD)

            dl=torch.utils.data.DataLoader(neg_dataset, batch_size=batch_size)
            d_logits_neg=[]
            for batch_index, (img, label) in enumerate(dl):
                # d_out_neg = D_neg.run(img.numpy(), is_training=False, minibatch_size=batch_size)
                if img.shape[0] % batch_size !=0:
                    break
                with sess.as_default():
                    d_out_neg = D_neg.get_output_for(tf.convert_to_tensor(img.numpy()), is_training=False).eval()
                d_logits_neg.extend(list(d_out_neg))
            d_scores_neg=sigmoid(torch.tensor(d_logits_neg))
            num_correct_neg=len(np.where(d_scores_neg>=0.5)[0])
            d_acc=num_correct_neg*1.0/len(neg_dataset)
`

Thanks.

**Comment by zsyzzsoft** [comment]
I'm not sure about your data pipeline, but note that the real images should be converted from [0, 255] to [-1, 1] before being fed into the discriminator.

**Comment by slala2121** [comment]
Thanks for your response. My real data was normalized to [0,1] and they were in this range for training as well. Is that an issue when computing the discriminator performance on the real data?

**Comment by zsyzzsoft** [comment]
It's not an issue if you can make sure that it's consistent across training and testing, but I guess that there must be something inconsistent. The discriminator should have the ability to classify the training images as real.

**Comment by slala2121** [comment]
Makes sense; actually I realized the data was being mapped from [0,1] to [-1,1] to I made that fix. 

Also were the training and validation accuracies computed on the real dataset only? or also synthetic images?

**Comment by zsyzzsoft** [comment]
We compute the training and validation accuracies on real images only.

**Comment by slala2121** [comment]
Ok, thanks
